### PR TITLE
Add a github action for generating code.

### PR DESCRIPTION
**Pull request by brendandburns**
cc @ityuhui 

The output of this github action is a pull request like:
~https://github.com/brendandburns/c/pull/3~
https://github.com/brendandburns/c/pull/5

### CODE REVIEW

File: .github/workflows/generate.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      - name: Generate Openapi
+        run: |
+          pushd gen/openapi
+          cat <<"EOF"> settings
+          # Kubernetes branch/tag to get the OpenAPI spec from.
+          export KUBERNETES_BRANCH="${{ github.event.inputs.kubernetesBranch }}"
+
+          # client version for packaging and releasing. It can
+          # be different than SPEC_VERSION.
+          export CLIENT_VERSION="0.3.0"
```
Comment (ityuhui) [inline .github/workflows/generate.yml:41]: The `CLIENT_VERSION` is passed to openapi-generator. After the code is generated, it's troublesome to modify. Can we make it configuratable e.g. workflow_dispatch.inputs ?

**Comment by ityuhui** [comment]
Hi @brendandburns 

There is another version setting in the file https://github.com/kubernetes-client/c/blob/master/kubernetes/PreTarget.cmake#L1-L3

```CMake
set(PROJECT_VERSION_MAJOR 0)
set(PROJECT_VERSION_MINOR 3)
set(PROJECT_VERSION_PATCH 0)
```

Should it be updated before releasing or via this action ? 
I can add a manul step to https://github.com/kubernetes-client/c/blob/master/RELEASE.md to address it.

---

And it's better to delete the old model/api/unit-test files in the C client before re-generating because some of them are deprecated in the new Kubernetes spec:

```bash
find $CLIENT_REPO_ROOT/kubernetes/model -type f -not -name "int_or_string*" -exec rm -r {} \;
find $CLIENT_REPO_ROOT/kubernetes/api -type f -not -name "int_or_string*" -exec rm -r {} \;
find $CLIENT_REPO_ROOT/kubernetes/unit-test -type f -not -name "manual*" -exec rm -r {} \;
```

File: .github/workflows/generate.yml (hunk 1 above)
Comment (ityuhui) [reply .github/workflows/generate.yml:41]: I checked out the code and found the `CLIENT_VERSION` is not used by the C generator.  So we can update it before releasing (Add a manul step to https://github.com/kubernetes-client/c/blob/master/RELEASE.md).

I have no problem here.

Comment (brendandburns) [reply .github/workflows/generate.yml]: Fixed this.

**Comment by brendandburns** [comment]
I've addressed your comments, I figured out a way to integrate the version update into the generation workflow.

Please take another look.

**Comment by ityuhui** [comment]
Thank you @brendandburns 

Now we have a great automated generation solution that saves time for upgrading to new versions and avoids mistakes.

**Pull request merged**